File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.py
from collections import defaultdict
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        dic = defaultdict(int)
        answer = []
        for i in range(len(nums)):
            dic[nums[i]]+=1
        heap = []
        heapq.heapify_max(heap)
        for _ in range(len(dic)):
            heapq.heappush(heap, dic.popitem()[::-1])
        heapq.heapify_max(heap)

        for _ in range(k):
            answer.append(heapq.heappop_max(heap)[1])
            heapq.heapify_max(heap)
        return answer
        

File: Data Structures & Algorithms/rotate-array/submission-3.go
func rotate(nums []int, k int) {
	j:=len(nums)-1
	k=k%len(nums)
	for i:=0; i<j; i++{
		nums[i], nums[j] = nums[j], nums[i]
		j--
	}
	j=k-1
	for i:=0; i<j; i++{
		nums[i], nums[j] = nums[j], nums[i]
		j--
	}
	j=len(nums)-1
	for i:=k; i<j; i++{
		nums[i], nums[j] = nums[j], nums[i]
		j--
	}
}
